ci/ansible/playbooks/os/rhel/upgrade-release.yml

```diff
@@ -0,0 +1,558 @@
+            patterns: '*.repo'
+          register: find_repo_files
+
+        - name: Set provider
+          set_fact:
+            provider: >-
+              {{ 'aws' if _repo_file_paths | select('search', leapp_dependencies.repos.aws.0.repo_file) else
+                 'azure' if _repo_file_paths | select('search', leapp_dependencies.repos.azure.0.repo_file) else
+                 'non_cloud' }}
+          vars:
+            _repo_file_paths: "{{ find_repo_files.files | map(attribute='path') }}"
+
+        - name: Print detected provider
+          debug:
+            var: provider
+
+    - name: Register SELinux state
+      set_fact:
+        pre_upgrade_selinux_facts: "{{ ansible_facts.selinux }}"
+
+    - name: Register enabled repositories
+      command: yum repolist --quiet
+      register: pre_upgrade_enabled_repositories
+      changed_when: false
+
+    - name: Ensure repositories that provide leapp utility are enabled
+      ini_file:
+        path: "{{ item.repo_file }}"
+        section: "{{ item.name }}"
+        option: enabled
+        value: 1
+        mode: u=rw,go=r
+      loop: "{{ leapp_dependencies.repos[provider] }}"
+
+    - name: Update repository certificates
+      when: provider == "azure"
+      yum:
+        enablerepo: rhui-microsoft-azure-rhel7
+        disablerepo: "*"
+        state: latest
+        update_only: true
+        name: "*"
+
+    - &UPDATE_ALL_PACKAGES
+      name: Update all packages in current major version
+      yum:
+        update_cache: true
+        name: "*"
+        state: latest  # noqa: package-latest
+
+    - &REBOOT_SYSTEM_AFTER_UPDATE
+      name: Reboot system after update  # to load kernel from latest minor version if any
+      reboot:
+        msg: Reboot initiated by Ansible due to update
+        connect_timeout: "{{ reboot.connect_timeout }}"
+        reboot_timeout: "{{ reboot.update_timeout }}"
+      vars:
+        # disable connection sharing to avoid random warning on stderr: Failed to reset connection
+        ansible_control_path: none
+
+    - &REFRESH_ANSIBLE_FACTS
+      name: Refresh Ansible facts
+      setup:
+        gather_subset: min
+
+    - name: Assert latest minor version
+      assert:
+        that: ansible_distribution_version is version(versions.required.full, '>=')
+        quiet: true
+
+    - name: Install packages that provide the leapp utility
+      package:
+        name: "{{ leapp_dependencies.packages[provider] }}"
+        state: present
+
+    - name: Copy leapp metadata archive
+      copy:
+        src: "{{ leapp_archive }}"
+        dest: /etc/leapp/files/leapp-data15.tar.gz
+        mode: preserve
+
+    - name: Unarchive leapp metadata
+      unarchive:
+        src: /etc/leapp/files/leapp-data15.tar.gz
+        dest: /etc/leapp/files/
+        remote_src: true
+
+    # Address upgrade inhibitors
+    # PermitRootLogin option is required in sshd_config file
+    - name: Ensure PermitRootLogin is defined
```

Comment:
We are doing this and not reverting after upgrade? Please also provide more comments on this here.

Reply:
This is by purpose since `PermitRootLogin yes` is considered as bad practice. If defined explicitly, playbook will preserve it, otherwise it will apply more secure setting. Actually this is not so important since Epiphany overwrites this file in `common` role.